index.js

```diff
@@ -112,6 +125,40 @@ module.exports = {
           });
       }
     };
+
+    //Adding event listeners to take automatic screenshot
+
+    if(autoCaptureOptions.indexOf('click') !== -1){
+      
+      nemo.driver.flow_.on('scheduleTask',function(task){
+        if(task !== undefined){
+          if(task.indexOf('WebElement.')!== -1){
+            var app = task.split('.');
+            if(app[1].indexOf('click')!== -1){
+              var path = screenShotPath;
+              var filename = 'ScreenShot-' + process.pid + '-' + new Date().getTime();
+              var screenShotFileName = path + '\\' + filename;
+              nemo.screenshot.snap(screenShotFileName);
+            }
+          }
+        }
+      });
+
+    }
+
+    if(autoCaptureOptions.indexOf('exception') !== -1){
+
+      nemo.driver.flow_.on('uncaughtException',function(exception){
+        var path = screenShotPath;
+        var filename = 'ScreenShot-' + process.pid + '-' + new Date().getTime();
+        var screenShotFileName = path + '\\' + filename;
+        nemo.screenshot.snap(screenShotFileName).then(function(){
```

Comment:
The four lines in both event based screenshots are identical. Please pull them up into a private method such as:
`eventScreenShot(eventID)`